bedrock/base/templates/macros.html

```diff
@@ -330,21 +330,21 @@
           <div>
             <p itemprop="articleBody">{{ format_tweet_body(tweet)|safe }}</p>
             {% if retweet -%}
-              <p class="retweet-credit">{{ _('Retweeted by %s')|format('<a href="%s" class="credit">%s</a>'|format('https://twitter.com/'+_tweet.user.screen_name, _tweet.user.name)|safe) }}</p>
+              <p class="retweet-credit">{{ _('Retweeted by %s')|format('<a href="%s" class="credit" data-social-network="twitter-only" data-social-action="Retweet Credit Link Exit">%s</a>'|format('https://twitter.com/'+_tweet.user.screen_name, _tweet.user.name)|safe) }}</p>
             {% endif -%}
             {% if tweet.entities.media -%}
               {% for medium in tweet.entities.media -%}
                 {% if medium.type == 'photo' -%}
-                  <p class="media"><a href="{{ medium.expanded_url }}" class="image"><img src="{{ medium.media_url_https }}" alt="" itemprop="image"></a></p>
+                  <p class="media"><a href="{{ medium.expanded_url }}" class="image" data-social-network="twitter-only" data-social-action="Preview Image Exit"><img src="{{ medium.media_url_https }}" alt="" itemprop="image"></a></p>
                 {% endif -%}
               {% endfor -%}
             {% endif -%}
           </div>
           <footer>
             <ul class="actions">
-              <li><a class="reply" href="https://twitter.com/intent/tweet?in_reply_to={{ tweet.id }}">{{ _('Reply') }}</a></li>
-              <li><a class="retweet" href="https://twitter.com/intent/retweet?tweet_id={{ tweet.id }}">{{ _('Retweet') }}</a></li>
-              <li><a class="favorite" href="https://twitter.com/intent/favorite?tweet_id={{ tweet.id }}">{{ _('Favorite') }}</a></li>
+              <li><a class="reply" href="https://twitter.com/intent/tweet?in_reply_to={{ tweet.id }}" data-social-network="twitter-only" data-social-action="Reply">{{ _('Reply') }}</a></li>
+              <li><a class="retweet" href="https://twitter.com/intent/retweet?tweet_id={{ tweet.id }}" data-social-network="twitter-only" data-social-action="Retweet">{{ _('Retweet') }}</a></li>
+              <li><a class="favorite" href="https://twitter.com/intent/favorite?tweet_id={{ tweet.id }}" data-social-network="twitter-only" data-social-action="Favorite">{{ _('Favorite') }}</a></li>
```

Comment:
It is more about the data-model than efficiency considerations.  Neither model affects efficiency much, but both quality of data and the flexibility for marketing decisions can be significantly increased thanks to small changes in the data-model.